python/essex-config/essex_config/config.py

```diff
@@ -0,0 +1,152 @@
+"""Main configuration module for the essex_config package."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from functools import cache
+from typing import Protocol, TypeVar, cast
+
+from pydantic import BaseModel
+from pydantic_core import PydanticUndefined
+
+from essex_config.sources import EnvSource, Source
+from essex_config.sources.source import Alias
+
+DEFAULT_SOURCE_LIST: list[Source] = [EnvSource()]
+
+
+T = TypeVar("T", bound=BaseModel)
+V = TypeVar("V", covariant=True)
+
+
+@dataclass
+class Prefixed:
+    """Class to define the prefix for the configuration."""
+
+    prefix: str
+
+
+class Config(Protocol[V]):
+    """Protocol to define the configuration class."""
+
+    @classmethod
+    def config_load(
```

Comment:
It's a little weird to have `config_load` call `load_config` - can we call these both `load_config`?

Reply:
changed!